migrations/truffle/1_2_witnet_requests.js

```diff
@@ -33,56 +32,73 @@ async function deployWitnetRequests (from, isDryRun, ecosystem, network, request
     const request = requests[key]
     if (request?.retrievals || request?.template) {
       const targetAddress = addresses[ecosystem][network].requests[key] ?? null
-      if (selection.length == 0 || selection.includes(key)) {
-        if (isDryRun || targetAddress === "") {
-          let requestAddress
-          if (request?.retrievals) {
-            try {
-              requestAddress = await deployWitnetRequest(from, key, request)
-            } catch (e) {
-              utils.traceHeader(`Failed '${key}': ${e}`)
-              process.exit(1)
-            }
-          } else {
-            try {
-              let templateAddr = findArtifactAddress(addresses[ecosystem][network], request?.template)
-              if (
-                utils.isNullAddress(templateAddr) ||
-                  (await web3.eth.getCode(templateAddr)).length <= 3
-              ) {
-                const templateArtifact = findTemplateArtifact(witnet.templates, request?.template)
-                if (!templateArtifact) {
-                  throw `artifact '${request?.template} not found in templates file`
-                }
-                templateAddr = await deployWitnetRequestTemplate(from, request?.template, templateArtifact)
-                if (utils.isNullAddress(templateAddr)) {
-                  throw `artifact '${request?.template}' could not get deployed`
-                }
-                addresses[ecosystem][network].templates[request?.template] = templateAddr
+      if (
+        targetAddress === "" ||
+        (selection.length == 0 & isDryRun) ||
+        (selection.length > 0 && selection.includes(key))
```

Comment:
The first clause seems redundant:
```suggestion
       selection.includes(key)
```